tests/lib/hardware/openstack_libcloud.py

```diff
@@ -201,6 +201,10 @@ def get_ssh_ip(self):
         # NOTE(jhesketh): For now, just use the last floating IP
         return self._floating_ips[-1].ip_address
 
+    def add_data_disk(self, capacity):
+        # TO BE IMPLEMENTED
```

Comment:
can you log a warning that this is not implemented yet?

Reply:
Done